mlrun/runtimes/sparkjob/abstract.py

```diff
@@ -715,6 +743,48 @@ def spec(self) -> AbstractSparkJobSpec:
     def spec(self, spec):
         raise NotImplementedError()
 
+    @staticmethod
+    def _verify_spark_job_limits(
+        resources_field_name,
+        cpu=None,
+    ):
+        if cpu:
+            verify_field_regex(
+                f"function.spec.{resources_field_name}.limits.cpu",
+                cpu,
+                mlrun.utils.regex.k8s_resource_quantity_regex,
+            )
+        return generate_resources(cpu=str(cpu))
+
+    @staticmethod
+    def _verify_spark_job_requests(
```

Comment:
Why duplicating the logic and not using the existing `_verify_and_set_limits` ?

Reply:
I've talked to @Tankilevitch about it... it is opposite than in other jobs - requests and limits has different parameters